interface.go

```diff
@@ -0,0 +1,96 @@
+package multiconfig
+
+import (
+	"errors"
+	"reflect"
+)
+
+// InterfaceLoader satisfies the loader interface. It recursively checks if a
+// struct implements the DefaultValues interface and applies the function
+// depth first, if it does
+// This is useful in case you want to overwrite default values set by tags
+// in a nested struct
+type InterfaceLoader struct {
+}
+
+type DefaultValues interface {
+	ApplyDefaults()
+}
+
+// structFields returns the exported fields of a struct value or pointer
+// returns nil if the input is not a struct
+func structFields(v reflect.Value) []*reflect.Value {
```

Comment:
I don't think it makes sense to work with `*reflect.Value`. All of it's methods and all reflect function work on `reflect.Value` directly